__tests__/patch-article-by-id-api.test.js

```diff
@@ -0,0 +1,79 @@
+const request = require("supertest");
+
+const db = require("../db/connection");
+const {
+  articleData,
+  commentData,
+  topicData,
+  userData,
+} = require("../db/data/test-data");
+const seed = require("../db/seeds/seed");
+const app = require("../app");
+
+beforeEach(() => seed({ articleData, commentData, topicData, userData }));
+afterAll(() => db.end());
+
+describe("PATCH /api/articles/:article_id", () => {
+  test("404: responds with an error msg if article does not exist", () => {
+    return request(app)
+      .patch("/api/articles/999")
+      .send({ inc_votes: 1 })
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("not found");
+      });
+  });
+  test("404: responds with an error msg if invalid inc_votes value", () => {
+    return request(app)
+      .patch("/api/articles/1")
+      .send({ inc_votes: "invalid" })
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Bad request");
+      });
+  });
+  test("201: responds with a updated article with a given article id", () => {
+    const newVote = {
+      inc_votes: 1,
+    };
+    return request(app)
+      .patch("/api/articles/1")
+      .send(newVote)
+      .expect(201)
+      .then(({ body }) => {
+        expect(body.article).toMatchObject({
+          article_id: 1,
+          title: "Living in the shadow of a great man",
+          topic: "mitch",
+          author: "butter_bridge",
+          body: "I find this existence challenging",
+          created_at: expect.any(String),
+          votes: 101,
+          article_img_url:
+            "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700",
+        });
+      });
+  });
+  test("201: decrements voteCount if inc_votes is negative and responds with a updated article with a given article id", () => {
+    const newVote = {
+      inc_votes: -100,
+    };
+    return request(app)
+      .patch("/api/articles/1")
+      .send(newVote)
+      .expect(201)
+      .then(({ body }) => {
+        expect(body.article).toMatchObject({
+          article_id: 1,
+          title: "Living in the shadow of a great man",
+          topic: "mitch",
+          author: "butter_bridge",
+          body: "I find this existence challenging",
+          created_at: expect.any(String),
+          votes: 0,
+          article_img_url:
+            "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700",
+        });

```

Comment:
Just a missing test to look at what happens if patching to a bad id, so patch /api/articles/banana Your error handler will probably already catch it, but good to have a solid test for it.